app/__init__.py

```diff
@@ -387,14 +387,17 @@ def checkout(dbapi_connection, connection_record, connection_proxy):
 
         @event.listens_for(db.engine, 'checkin')
         def checkin(dbapi_connection, connection_record):
-            # connection returned by a web worker
-            TOTAL_CHECKED_OUT_DB_CONNECTIONS.dec()
-
-            # duration that connection was held by a single web request
-            duration = time.monotonic() - connection_record.info['checkout_at']
-
-            DB_CONNECTION_OPEN_DURATION_SECONDS.labels(
-                connection_record.info['request_data']['method'],
-                connection_record.info['request_data']['host'],
-                connection_record.info['request_data']['url_rule']
-            ).observe(duration)
+            try:
+                # connection returned by a web worker
+                TOTAL_CHECKED_OUT_DB_CONNECTIONS.dec()
+
+                # duration that connection was held by a single web request
+                duration = time.monotonic() - connection_record.info['checkout_at']
+
+                DB_CONNECTION_OPEN_DURATION_SECONDS.labels(
+                    connection_record.info['request_data']['method'],
+                    connection_record.info['request_data']['host'],
+                    connection_record.info['request_data']['url_rule']
+                ).observe(duration)
+            except Exception:
+                current_app.logger.exception("Exception caught for checkin event.")
```

Comment:
Can we log the original exception message so we have some context if we see this?

Reply:
logger.exception will do that

Reply:
Ah OK, thanks